### ISSUE TITLE
[Phase 4] Classical ML Models

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/ecg-arrhythmia-classification#5]
## Phase 4: Classical ML Models

**Parent:** #1
**Depends on:** #4

### Objectives
Train and evaluate classical ML models (Random Forest, XGBoost, SVM) on extracted features.

### Tasks

- [ ] Implement Random Forest classifier with class weights
- [ ] Implement XGBoost classifier
- [ ] Implement SVM classifier (RBF kernel)
- [ ] Create 5-fold stratified cross-validation pipeline
- [ ] Handle class imbalance (SMOTE or class weights)
- [ ] Implement hyperparameter tuning (grid search)
- [ ] Save best models with joblib
- [ ] Log training metrics

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/models.py` | Create | Model definitions |
| `src/train.py` | Create | Training pipeline |
| `models/` | Populate | Saved model weights |

### Code Reference

```python
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from xgboost import XGBClassifier
from sklearn.model_selection import StratifiedKFold, cross_val_score
from sklearn.preprocessing import StandardScaler
import joblib

def get_classical_models():
    return {
        'RandomForest': RandomForestClassifier(
            n_estimators=200,
            max_depth=20,
            class_weight='balanced',
            n_jobs=-1,
            random_state=42
        ),
        'XGBoost': XGBClassifier(
            n_estimators=200,
            max_depth=6,
            learning_rate=0.1,
            use_label_encoder=False,
            eval_metric='mlogloss',
            random_state=42
        ),
        'SVM': SVC(
            kernel='rbf',
            C=10,
            gamma='scale',
            class_weight='balanced',
            probability=True,
            random_state=42
        )
    }

def train_with_cv(X, y, model, n_splits=5):
    scaler = StandardScaler()
    X_scaled = scaler.fit_transform(X)

    skf = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=42)
    scores = cross_val_score(model, X_scaled, y, cv=skf, scoring='f1_weighted')

    print(f"CV F1 scores: {scores}")
    print(f"Mean F1: {scores.mean():.4f} (+/- {scores.std():.4f})")

    # Train final model on all data
    model.fit(X_scaled, y)
    return model, scaler
```

### Definition of Done

- [ ] All three models train without errors
- [ ] Cross-validation F1 scores reported
- [ ] Class imbalance handled (check minority class recall)
- [ ] Best model saved to `models/` directory
- [ ] Training logs show convergence

### Expected Results

| Model | Accuracy | F1 (weighted) |
|-------|----------|---------------|
| Random Forest | 95-97% | 0.93-0.95 |
| XGBoost | 96-98% | 0.94-0.96 |
| SVM | 94-96% | 0.92-0.94 |

### Technical Notes

**For junior developers:**
- `class_weight='balanced'` adjusts for imbalanced classes
- Always scale features before SVM (sensitive to scale)
- XGBoost handles imbalance with `scale_pos_weight` for binary, use sample weights for multi-class
- Use `n_jobs=-1` for parallel training on all CPU cores
